### PR TITLE
Fix quad mesh cursor data

### PR DESCRIPTION
**Pull request by raphaelquast**
## PR Summary

Closes: #22908 

This is just a very minor bug I noticed...

When using a `QuadMesh` **without** setting an array, the `get_cursor_data` function throws an error because it does not check if an array has been set.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [x] New features are documented, with examples if plot related.
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/collections.py [hunk 1]
```diff
@@ -2201,6 +2201,7 @@ def get_cursor_data(self, event):
         contained, info = self.contains(event)
         if len(info["ind"]) == 1:
             ind, = info["ind"]
-            return self.get_array()[ind]
+            array = self.get_array()
+            return self.get_array()[ind] if array else None
```
Comment (QuLogic) [inline lib/matplotlib/collections.py:2205]: Should use `array` instead of calling `get_array` again.

Comment (raphaelquast) [reply lib/matplotlib/collections.py]: yep sure! don't know why I didn't do that in the first place...

**Comment by oscargus** [comment]
I took the liberty of editing your post to point out the issue number. In this way, Github links this PR and the issue and the issue will be automatically closed when this is merged.

**Review by oscargus — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
meeseeksdev backport to v3.5.x

**Comment by QuLogic** [comment]
Thanks @raphaelquast! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Comment by greglucas** [comment]
I don't think this was quite right because `get_array()` often returns a numpy array, which when evaluated with `if array` raises a ValueError.

Hover the cursor over the figure in this example:
```python
import matplotlib.pyplot as plt
import numpy as np
n = 4
x = np.arange(n)
X = x[:, None] * x[None, :]

fig, ax = plt.subplots()
mesh = ax.pcolormesh(X)
mesh.set_mouseover(True)

plt.show()
```

**Comment by raphaelquast** [comment]
@greglucas sorry... indeed that's not the desired behavior!

I opened a new pull request to fix this. (#22957)
- `if array` >> `if array is not None`
... this should work as expected.

**Pull request merged**